app/lib/task/scheduler.dart

```diff
@@ -206,8 +207,40 @@ Future<void> schedule(
             arguments: [payload!],
             description: description!,
           );
+        } on ZoneExhaustedException catch (e, st) {
+          // A zone being exhausted is normal operations, we just use another
+          // zone for 15 minutes.
+          _log.info(
+            'zone resources exhausted, banning ${e.zone} for 30 minutes',
+            e,
+            st,
+          );
+          // Ban usage of zone for 30 minutes
+          zoneBannedUntil[e.zone] = clock.now().add(Duration(minutes: 30));
+
+          rollbackPackageState = true;
+        } on QuotaExhaustedException catch (e, st) {
+          // Quota exhausted, this can happen, but it shouldn't. We'll just stop
+          // doing anything for 10 minutes. Hopefully that'll resolve the issue.
+          // We log severe, because this is a reason to adjust the quota or
+          // instance limits.
+          _log.severe(
+            'Quota exhausted trying to create $instanceName, banning all zones '
+            'for 10 minutes',
+            e,
+            st,
+          );
+
+          // Ban all zones for 10 minutes
+          for (final zone in compute.zones) {
+            zoneBannedUntil[zone] = clock.now().add(Duration(minutes: 10));
```

Comment:
This may override a zone ban with ~30 minutes from line#219. Maybe move it behind a method that checks the existence of a ban and sets the later in time.

Reply:
Great catch...